src/app/blogs/page.jsx

```diff
@@ -1,16 +1,73 @@
-import { allBlogs } from '@/queries/blogs';
-import Blogs from '@/components/ui/Blogs';
+import BlogItem from '@/components/ui/BlogItem';
+import PaginationControls from '@/components/ui/PaginationControls';
 import { fetchGraphQL } from '@/lib/graphqlUtils';
-import Link from 'next/link';
+import { allBlogs, latestBlogs } from '@/queries/blogs';
 
-export default async function BlogsPage() {
-  const { blogs } = await fetchGraphQL(allBlogs, { next: { tags: ['blogs'] } });
+const Blogs = async ({ searchParams }) => {
+  const { blogsPage } = await fetchGraphQL(allBlogs, {
+    next: { tags: ['blogs'] },
+  });
+  const { blogsPage: latestBlogsPage } = await fetchGraphQL(latestBlogs, {
+    next: { tags: ['blogs'] },
+  });
+  const { chapterTextForAllBlogs, chapterTextForLatest, blogPosts } = blogsPage;
+  const { blogPosts: latestBlogPosts } = latestBlogsPage;
+
+  const perPage = 5;
+
+  const page = searchParams['page'] || '1';
```

Comment:
Simple solution for pagination, but I would recommend taking a look at cursor pagination:

https://www.youtube.com/watch?v=HZRWqlGEph4&ab_channel=AdamRichardson

Also this solution you use doesn't seem like it can be reused for a different list easily, if we had another place where we need pagination you would have to repeat all of this code

Reply:
https://www.youtube.com/watch?v=8FYvmFSKtC4&ab_channel=Hygraph